src/EditorFeatures/Core/LanguageServer/AbstractInProcLanguageClient.cs

```diff
@@ -43,6 +44,20 @@ internal abstract partial class AbstractInProcLanguageClient : ILanguageClient,
         /// </summary>
         public string Name => ServerKind.ToUserVisibleString();
 
+        /// <summary>
+        /// Gets the optional middle layer object that can intercept outgoing requests and responses.
+        /// </summary>
+        /// <remarks>
+        /// Currently utilized by Razor to intercept Roslyn's workspace/semanticTokens/refresh requests.
+        /// </remarks>
+        public object? MiddleLayer => _middleLayer;
```

Comment:
Interesting - why do we need to support a middle layer here?  Shouldn't that be possible entirely on the client side?

Reply:
Interesting.  It seems like it could be done via content type or lsp server name.  But I dont generally have a problem with this, was just wondering